### ISSUE TITLE
bug(app-check): debug provider cannot be used

### ISSUE THREAD

**Issue by robingenz** [capawesome-team/capacitor-firebase#453]
### Plugin(s)

- [ ] Analytics
- [ ] App
- [X] App Check
- [ ] Authentication
- [ ] Crashlytics
- [ ] Cloud Firestore
- [ ] Cloud Messaging
- [ ] Performance
- [ ] Remote Config

### Did you test the latest version?

- [X] I use the latest version

### Platform(s)

- [ ] Android
- [X] iOS
- [ ] Web

### Current behavior

When retrieving a token with the debug provider, you get the following error message.

![image](https://github.com/robingenz/capacitor-firebase-plugin-demo/assets/43593052/d6a4b0ab-7570-4827-8be4-421e01768ec6)

See https://github.com/robingenz/capacitor-firebase-plugin-demo/issues/299

### Expected behavior

I expected to successfully retrieve the Firebase App Check token without any errors.

### Reproduction

https://github.com/robingenz/capacitor-firebase-plugin-demo

### Steps to reproduce

1. Open the iOS app.
2. Navigate to the screen where the Firebase App Check token is requested.
3. Check "Use Debug Provider" and click on the "Initialize" button.
4. Click on the "Get Token" button.
5. Observe the error message.

### Other information

Until now, Firebase on Android and iOS is initialized automatically when the app is launched, see [this example](https://github.com/capawesome-team/capacitor-firebase/blob/f845e3c590640119f00cadb804ebac971c8f102a/packages/messaging/ios/Plugin/FirebaseMessaging.swift#L17).
However, sometimes Firebase may not be initialized directly at startup. This is the case for Firebase App Check if you want to use the debug provider, see [here](https://firebase.google.com/docs/app-check/ios/debug-provider).

```swift
let providerFactory = AppCheckDebugProviderFactory()
AppCheck.setAppCheckProviderFactory(providerFactory)

FirebaseApp.configure()
```

### Capacitor doctor

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 5.4.2
  @capacitor/core: 5.4.2
  @capacitor/android: 5.4.2
  @capacitor/ios: 5.4.2

Installed Dependencies:

  @capacitor/cli: 5.4.2
  @capacitor/core: 5.4.2
  @capacitor/android: 5.4.2
  @capacitor/ios: 5.4.2

[success] iOS looking great! 👌
[success] Android looking great! 👌

### Before submitting

- [X] I understand that incomplete issues (e.g. without reproduction) are closed.

**Comment by ludufre** [comment]
Any progress on this issue?

**Comment by robingenz** [comment]
No, we need to refactor all plugins to be able to resolve this issue.

**Comment by eljass** [comment]
Workaround for this one in iOS we found is to set AppCheck debug earlier than you call first `AppCheck.configure()`. We do this in our AppDelegate.swift.

```swift
#if DEBUG
let isDebugEnabled = ProcessInfo.processInfo.arguments.contains("-FIRDebugEnabled")
if isDebugEnabled {
    AppCheck.setAppCheckProviderFactory(AppCheckDebugProviderFactory())
    CAPLog.print("\(AppDelegate.TAG): AppCheck: Debug mode enabled, AppCheck debug provider factory set.")
} else {
    CAPLog.print("\(AppDelegate.TAG): AppCheck: Debug mode disabled, let AppCheck plugin initialize normally.")
}
#endif

// Then you can either allow plugins to call FirebaseApp.configure() or if you need it in your own native code, call it after the block above
```

Good to remember that currently setting `debug` or `debugToken` in `FirebaseAppCheck.initialize()` call, it will not work. So good to have it like this if you use it on other platforms:
`await FirebaseAppCheck.initialize({ debugToken: Capacitor.getPlatform() !== 'ios' ? true : false })`

**Comment by eljass** [comment]
@robingenz and any others inerested. I've added proposal for this workarond for the current limitations before this plugin is getting bigger refactoring.

I do also feel that only refactoring these plugins is not enough for iOS as many developers might configure Firebase app in their own native code as well (at least that was the case on our side).

Feel free to refer the code and updated readme, how I got this plugin's debug mode working correctly on iOS and Android.
#897
